cluster-autoscaler/cloudprovider/test/test_cloud_provider.go

```diff
@@ -170,24 +227,25 @@ func (tng *TestNodeGroup) IncreaseSize(delta int) error {
 	tng.targetSize += delta
 	tng.Unlock()
 
-	return tng.cloudProvider.onIncrease(tng.id, delta)
+	return tng.cloudProvider.onScaleUp(tng.id, delta)
 }
 
 // Exist checks if the node group really exists on the cloud provider side. Allows to tell the
 // theoretical node group from the real one.
 func (tng *TestNodeGroup) Exist() bool {
-	return true
+	return tng.exist
 }
 
 // Create creates the node group on the cloud provider side.
 func (tng *TestNodeGroup) Create() error {
-	return cloudprovider.ErrAlreadyExist
+	tng.cloudProvider.AddAutoprovisionedNodeGroup(tng.id, tng.minSize, tng.maxSize, 0, tng.machineType)
```

Comment:
Can we verify here (or in AddAutoporovisionedNodeGroup) that the node group doesn't already exist first? If we just overwrite it in tests we may miss some bugs related to avoiding/handling this case.

Reply:
Done.